solidity/random-beacon/tasks/initialize.ts

```diff
@@ -0,0 +1,285 @@
+import { task, types } from "hardhat/config"
+
+import type { BigNumber } from "ethers"
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
+
+const TASK_INITIALIZE = "initialize"
+const TASK_MINT = `${TASK_INITIALIZE}:mint`
+const TASK_STAKE = `${TASK_INITIALIZE}:stake`
+const TASK_AUTHORIZE = `${TASK_INITIALIZE}:authorize`
+const TASK_REGISTER = `${TASK_INITIALIZE}:register`
+
+task(TASK_INITIALIZE, "Initializes staking for an operator")
+  .addParam("owner", "Stake Owner address", undefined, types.string)
+  .addParam("provider", "Staking Provider", undefined, types.string)
+  .addParam("operator", "Staking Operator", undefined, types.string)
+  .addOptionalParam("beneficiary", "Stake Beneficiary", undefined, types.string)
+  .addOptionalParam("authorizer", "Stake Authorizer", undefined, types.string)
+  .addOptionalParam("amount", "Stake amount", 1_000_000, types.int)
+  .addOptionalParam(
+    "authorization",
+    "Authorization amount (default: minimumAuthorization)",
+    undefined,
+    types.int
+  )
+  .addOptionalParam("force", "Force initialization", false, types.boolean)
+  .setAction(async (args, hre) => {
+    await initialize(hre, args)
+  })
+
+async function initialize(
+  hre: HardhatRuntimeEnvironment,
+  args: {
+    owner: string
+    provider: string
+    operator: string
+    beneficiary: string
+    authorizer: string
+    amount: BigNumber
+    authorization: BigNumber
+    force: boolean
+  }
+) {
+  if (await isAlreadyStaked(hre, args.operator)) {
+    console.log(`Operator ${args.operator} is already staked`)
```

Comment:
Operator ${args.operator} has already staked ?

Reply:
I'm not sure about that and renaming `isAlreadyStaked` to `hasAlreadyStaked`. We pass an operator as an argument. It's not the operator who stakes, so `hasAlreadyStaked` is not the best fit in my opinion. Maybe a better name would be `wasAlreadyStaked` or `hasAlreadyBeenStaked`?

Reply:
I see.. in this case up to you. I would probably rename `isAlreadyStaked` -> `isOperatorStaked` and put in the logs smth like "Operator {operator} has been staked already".